Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -2660,8 +2669,413 @@ def fetch_incidents():
             current_fetch_info_idp_detections = updated_last_run
             demisto.debug(f"CrowdstrikeFalconMsg: Ending fetch idp_detections. Fetched {len(idp_detections)}")
 
-    demisto.setLastRun([current_fetch_info_detections, current_fetch_info_incidents, current_fetch_info_idp_detections])
-    return incidents + detections + idp_detections
+    if 'Indicator of Misconfiguration' in fetch_incidents_or_detections:
+        demisto.debug('Fetching Indicator of Misconfiguration incidents')
+        demisto.debug(f'{current_fetch_info_iom_configs=}')
+        first_fetch_timestamp = format_time_to_custom_date_format(
+            time=FETCH_TIME,
+            date_format=IOM_DATE_FORMAT,
+            dateparser_settings={'TIMEZONE': 'UTC', 'RETURN_AS_TIMEZONE_AWARE': True})
+        new_scan_time = last_scan_time = current_fetch_info_iom_configs.get('last_scan_time', first_fetch_timestamp)
+        # The offset is used when not all the results have been returned from the API, therefore,
+        # we would need to do pagination using the offset query parameter
+        iom_next_token = current_fetch_info_iom_configs.get('iom_next_token')
+        # In order to deal with duplicates, we retrieve the last resource ids of the last run, so we can
+        # compare them with the newly fetched ids, and ignore any duplicates
+        last_fetch_resource_ids: list[str] = current_fetch_info_iom_configs.get('last_resource_ids', [])
+        filter = 'scan_time:'
+        if iom_next_token:
+            # If entered here, that means we are currently doing pagination, and we need to use the
+            # same fetch query as the previous round
+            filter = current_fetch_info_iom_configs.get('last_fetch_filter', '')
```

Comment:
If we are doing pagination, the filter should **not** be empty, since the API requires the same filter when doing pagination. Do you think we should add a validation, and if it is empty, raise an error?

Reply:
Concerning the API, the filter can be empty, but it shouldn't be empty in our code when we are doing pagination.

Reply:
yeah, let's add a check for it

Reply:
Added to the [integration](https://github.com/demisto/content/pull/29716/files#diff-8e2201c0967d2f600afbeaf2c976d0a14e05b53f1c18be2bc674e53d3b9f30bbR2778), and [unit test](https://github.com/demisto/content/pull/29716/files#diff-e6c664fe14b362fb816dce412eb0a3f1ae575df118a284960f7b0df660951d8fR6035)